/workspace$ flask test user
=========================== test session starts ===========================
platform linux -- Python 3.11.13, pytest-7.0.1, pluggy-1.6.0
rootdir: /home/runner/workspace, configfile: pytest.ini, testpaths: App/tests
plugins: typeguard-4.3.0
collected 8 items                                                         

App/tests/test_app.py::UserUnitTests::test_check_password PASSED    [ 12%]
App/tests/test_app.py::UserUnitTests::test_get_json PASSED          [ 25%]
App/tests/test_app.py::UserUnitTests::test_hashed_password FAILED   [ 37%]
App/tests/test_app.py::UserUnitTests::test_new_user PASSED          [ 50%]
App/tests/test_app.py::test_authenticate PASSED                     [ 62%]
App/tests/test_app.py::UsersIntegrationTests::test_create_user PASSED [ 75%]
App/tests/test_app.py::UsersIntegrationTests::test_get_all_users_json PASSED [ 87%]
App/tests/test_app.py::UsersIntegrationTests::test_update_user PASSED [100%]

================================ FAILURES =================================
___________________ UserUnitTests.test_hashed_password ____________________

self = <App.tests.test_app.UserUnitTests testMethod=test_hashed_password>

    def test_hashed_password(self):
        password = "mypass"
>       hashed = generate_password_hash(password, method='sha256')

App/tests/test_app.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.pythonlibs/lib/python3.11/site-packages/werkzeug/security.py:108: in generate_password_hash
    h, actual_method = _hash_internal(method, salt, password)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

method = 'sha256', salt = 'JCgb1A6S6BJE26g4', password = 'mypass'

    def _hash_internal(method: str, salt: str, password: str) -> tuple[str, str]:
        method, *args = method.split(":")
        salt_bytes = salt.encode()
        password_bytes = password.encode()
    
        if method == "scrypt":
            if not args:
                n = 2**15
                r = 8
                p = 1
            else:
                try:
                    n, r, p = map(int, args)
                except ValueError:
                    raise ValueError("'scrypt' takes 3 arguments.") from None
    
            maxmem = 132 * n * r * p  # ideally 128, but some extra seems needed
            return (
                hashlib.scrypt(
                    password_bytes, salt=salt_bytes, n=n, r=r, p=p, maxmem=maxmem
                ).hex(),
                f"scrypt:{n}:{r}:{p}",
            )
        elif method == "pbkdf2":
            len_args = len(args)
    
            if len_args == 0:
                hash_name = "sha256"
                iterations = DEFAULT_PBKDF2_ITERATIONS
            elif len_args == 1:
                hash_name = args[0]
                iterations = DEFAULT_PBKDF2_ITERATIONS
            elif len_args == 2:
                hash_name = args[0]
                iterations = int(args[1])
            else:
                raise ValueError("'pbkdf2' takes 2 arguments.")
    
            return (
                hashlib.pbkdf2_hmac(
                    hash_name, password_bytes, salt_bytes, iterations
                ).hex(),
                f"pbkdf2:{hash_name}:{iterations}",
            )
        else:
>           raise ValueError(f"Invalid hash method '{method}'.")
E           ValueError: Invalid hash method 'sha256'.

.pythonlibs/lib/python3.11/site-packages/werkzeug/security.py:70: ValueError
========================= short test summary info =========================
FAILED App/tests/test_app.py::UserUnitTests::test_hashed_password - Valu...
======================= 1 failed, 7 passed in 1.03s =======================
~/workspace$ ^C